pkg/sdk/errors.go

```diff
@@ -48,3 +47,230 @@ func decodeDriverError(err error) error {
+If any of the error information is missing in the issue body, please fill it up.
+Any additional information (or context what you were trying to achieve) would be helpful
+to provide the solution or fix as soon as possible. Thanks :)
+
+https://github.com//Snowflake-Labs/terraform-provider-snowflake/issues/new?labels=bug&title=%s&body=%s
+
+Errors:
+%s`,
+		url.QueryEscape("New issue"),
+		url.QueryEscape(fmt.Sprintf(ghIssueBodyTemplate, err.Error())),
+		err.Error(),
+	)
+}
+
+type SDKError struct { //nolint:all
+	file         string
+	line         int
+	message      string
+	nestedErrors []error
+}
+
+func (e *SDKError) errorFileInfoHidden() string {
+	builder := new(strings.Builder)
+	writeTree(e, builder, 0, false)
+	return builder.String()
+}
+
+func (e *SDKError) Error() string {
+	builder := new(strings.Builder)
+	writeTree(e, builder, 0, true)
+	return builder.String()
+}
+
+// NewError Creates new sdk.SDKError with information like filename or line number
+// depending on where NewError was called
+func NewError(message string) error {
+	return newSDKError(message, 2)
+}
+
+// NewPredefinedError Lets you predefine factory method for given sdk.SDKError which is convenient
+// when given error must be returned multiple times
+func NewPredefinedError(message string) func() error {
+	return func() error {
+		return newSDKError(message, 2)
+	}
+}
+
+type SDKPredefinedError func() error //nolint:all
+
+func (fn SDKPredefinedError) Error() string {
+	return fn().Error()
+}
+
+// NewPredefinedError2 same as NewPredefinedError, but we're getting filename + line where this error is declared,
+// not returned, which may be worse than the NewPredefinedError
+// version, but this one is better at replacing current predefined errors like ErrInvalidObjectIdentifier. With NewPredefinedError
+// we would have to change every return statement of this error to function call (because NewPredefinedError returns factory method for predefined err)
+// and the nature of NewPredefinedError2 is that we can return value as with ErrInvalidObjectIdentifier example.
+// The only disadvantage of this is that we'll get file + line from declaration which may cause issues that we have right now
+// (not knowing places where errors have been returned from - one function can return ErrInvalidObjectIdentifier in two places
+// and we'll know from which place with this approach unfortunately) and this error should solve them.
+// We can use both (e.g. for new errors use NewPredefinedError2 and for replacing old use NewPredefinedError)
+// or we can just choose one approach over the other. (of course NewPredefinedError2 will be renamed to something that makes more sense)
+// TODO this approach to predefined errors, this is lazy evaluated error, so the only filename + line number
+// TODO if we're going with this approach, change other methods to handle SDKPredefinedError type
+func NewPredefinedError2(message string) SDKPredefinedError {
```

Comment:
this is a terrible name. at least call it "NewPredfinedLegacyError" or something to that effect.

Reply:
Removed (according to discussions), so no changes applied